pkg/packagekit/autoupdates.jsx

```diff
@@ -337,6 +337,144 @@ class Dnf5Impl extends ImplBase {
+        } catch (error) {
+            console.error("AptImpl: getConfig failed:", error);
+            this.supported = false;
+        }
+    }
+
+    async setConfig(enabled, type, day, time) {
+        this.defaultConfigFile = "/etc/apt/apt.conf.d/50unattended-upgrades";
+        this.configFile = "/etc/apt/apt.conf.d/52unattended-upgrades-cockpit";
+
+        const timerConfD = "/etc/systemd/system/apt-daily-upgrade.timer.d";
+        const timerConf = timerConfD + "/time.conf";
+
+        let script = "set -e; ";
+
+        if (type !== null) {
+            // Configure Origins-Pattern based on update type in our override file
+            script += "cat > " + this.configFile + " << 'EOF'\n";
+            script += "// Cockpit managed configuration\n";
+            // Clear Unattended-Upgrade::Origins-Pattern as they get merged otherwise
+            script += "#clear Unattended-Upgrade::Origins-Pattern;\n";
+            script += "Unattended-Upgrade::Origins-Pattern {\n";
+            if (type === "security") {
+                // Only security updates
+                script += '        "origin=Debian,codename=${distro_codename},label=Debian-Security";\n';
+                script += '        "origin=Debian,codename=${distro_codename}-security,label=Debian-Security";\n';
+            } else {
+                // All updates (security + regular)
+                script += '        "origin=Debian,codename=${distro_codename},label=Debian";\n';
+                script += '        "origin=Debian,codename=${distro_codename},label=Debian-Security";\n';
+                script += '        "origin=Debian,codename=${distro_codename}-security,label=Debian-Security";\n';
+            }
+            script += "};\n\n";
+
+            // Enable Automatic reboot (to conform to current dnf implementation)
+            script += "// Enable automatic reboot when required\n";
+            script += 'Unattended-Upgrade::Automatic-Reboot "true";\n';
+            script += "EOF\n";
+        }
+
+        // Configure timer schedule
+        if (time !== null || day !== null) {
+            if (day === "" && time === "") {
+                // Restore defaults
+                script += "rm -f " + timerConf + "; ";
+            } else {
+                if (day == null)
+                    day = this.day;
+                if (time == null)
+                    time = this.time;
+                script += "mkdir -p " + timerConfD + "; ";
+                // Clear existing OnCalendar and set new one (like systemd override pattern)
+                script += "printf '[Timer]\\nOnCalendar=\\nOnCalendar=" + day + " " + time + "\\n' > " + timerConf + "; ";
+                script += "systemctl daemon-reload; ";
+            }
+        }
```

Comment:
This could become more generic and shared with the Dnf5Impl, but can be left as a follow up.